### PR TITLE
Eng 43 pagination

### PR DESCRIPTION
**Pull request by waseemanwar1**

### CODE REVIEW

**Review by prio101 — COMMENTED** [review]
If you can then disable the prev and next button if page has no tracks. ( If possible ).
Maybe you can use previousLinkClassName . or disabledClassName api for the react-paginate.

File: app/javascript/components/routes/Baked.js [hunk 1]
```diff
@@ -81,53 +85,91 @@ class Baked extends React.Component {
...
-	componentDidMount() {
-		this.getTracks();
+		this.handlePageClick = this
+            .handlePageClick
+            .bind(this);
 	}
 
 	getTracks = () => {
-		if (tracks.length > 0) return this.setState({ loading: false });
+		if (this.state.data.length > 0) return this.setState({ loading: false });
```
Comment (prio101) [inline app/javascript/components/routes/Baked.js:101]: Can we please assign this.state.data to tracks variable ? Not high priority , if you want you can let it be.

File: app/javascript/components/routes/Baked.js [hunk 2]
```diff
@@ -81,53 +85,91 @@ class Baked extends React.Component {
...
 
 		$.get('/baked_tracks').done((res) => {
-			tracks = res.baked_tracks;
-			this.setState({ loading: false });
+			const data = res.baked_tracks;
```
Comment (prio101) [inline app/javascript/components/routes/Baked.js:104]: I dont trust JS , you should not too... please assign the data into let rather than const.

**Review by prio101 — COMMENTED** [review]
Also if possible add the ADD TRACK component top of the list of the tracks thumbnails.
it goes  below everything. Maybe make a new small PR for that.

**Review by prio101 — COMMENTED** [review]
Please assure me that: 
- upon adding tracks as logged in user into backed page.
- it is showing into my artist page.
- but not showing into backed page.
image

artist page:

image

File: app/javascript/components/routes/Baked.js (hunk 1 above)
Comment (waseemanwar1) [reply app/javascript/components/routes/Baked.js:101]: I already tried with that approach, but its not working fine with that way.

**Review by prio101 — APPROVED** [review]

**Pull request merged**